core/src/wallet/tezos/TezosLikeAccount2.cpp

```diff
@@ -243,65 +243,82 @@ namespace ledger {
-                            tx->setType(request.type);
-                            const auto counterAddress = protocolUpdate == api::TezosConfigurationDefaults::TEZOS_PROTOCOL_UPDATE_BABYLON ?
-                                                        managerAddress : senderAddress;
-                            return explorer->getCounter(counterAddress).flatMapPtr<Block>(self->getContext(), [self, tx, explorer] (const std::shared_ptr<BigInt> &counter) {
-                                if (!counter) {
-                                    throw make_exception(api::ErrorCode::RUNTIME_ERROR, "Failed to retrieve counter from network.");
-                                }
-                                // We should increment current counter
-                                tx->setCounter(std::make_shared<BigInt>(++(*counter)));
-                                return explorer->getCurrentBlock();
-                            }).flatMapPtr<api::TezosLikeTransaction>(self->getContext(), [self, explorer, tx, senderAddress] (const std::shared_ptr<Block> &block) {
-                                tx->setBlockHash(block->hash);
-
-                                // Check whether we need a reveal operation
-                                // We assume that accounts are synced
-                                soci::session sql(self->getWallet()->getDatabase()->getPool());
-                                // Check if there is a send operation, otherwise we need a reveal
-                                uint64_t count = 0;
-                                sql << "SELECT COUNT(*) FROM tezos_transactions WHERE sender = :address", use(senderAddress), into(count);
-                                tx->reveal(count == 0);
-
-                                if (tx->getType() == api::TezosOperationTag::OPERATION_TAG_ORIGINATION) {
-                                    std::vector<TezosLikeKeychain::Address> listAddresses{self->_keychain->getAddress()};
-                                    return explorer->getBalance(listAddresses).mapPtr<api::TezosLikeTransaction>(self->getContext(), [tx] (const std::shared_ptr<BigInt> &balance) {
+                            // Check if recepient is revealed or not
+                            // because if not we have to add additional fees equal to storage_limit (in mXTZ)
+                            auto getRevealFee = [self, explorer, request]() -> Future<BigInt> {
+                               if (request.type != api::TezosOperationTag::OPERATION_TAG_TRANSACTION) {
+                                   return Future<BigInt>::successful(BigInt::ZERO);
+                               }
+                                return explorer->getManagerKey(request.toAddress).map<BigInt>(self->getContext(), [request] (const std::string &managerKey) -> BigInt {
+                                    // Base unit is uXTZ
+                                    return !managerKey.empty() ? BigInt::ZERO : *request.storageLimit * BigInt("1000");
```

Comment:
[OPT] BigInt(1000);? Why string?

Reply:
That was actually a good question I would have preferred to see resolved before merging. :( @pollastri-pierre